packages/wonder-blocks-tooltip/components/tooltip.js

```diff
@@ -177,18 +181,19 @@ export default class Tooltip extends React.Component<Props, State> {
     _renderTooltipAnchor(ids?: IIdentifierFactory) {
         const {forceAnchorFocusivity} = this.props;
         return (
-            <TooltipAnchor
-                forceAnchorFocusivity={forceAnchorFocusivity}
-                anchorRef={(r) => this._updateAnchorElement(r)}
-            >
-                {(active) => (
-                    <TooltipPortalMounter
-                        anchor={this._renderAnchorElement(ids)}
-                    >
-                        {this._renderPopper(active, ids)}
-                    </TooltipPortalMounter>
-                )}
-            </TooltipAnchor>
+            <React.Fragment>
+                <TooltipAnchor
+                    forceAnchorFocusivity={forceAnchorFocusivity}
+                    anchorRef={(r) => this._updateAnchorElement(r)}
+                    onActiveChanged={(active) => this.setState({active})}
+                >
+                    {this._renderAnchorElement(ids)}
+                </TooltipAnchor>
+                {this.state.active &&
+                    <TooltipPortalMounter anchorNode={this.state.anchorElement}>
+                        {this._renderPopper(this.state.active, ids)}
+                    </TooltipPortalMounter>}
+            </React.Fragment>
```

Comment:
I've reworked this some more so that there's no nesting at all between the tooltip or the portal.  React.Fragment allows us to do this would introducing any sort of wrapper component around the anchor except in the case of strings where it's actually needed.

Reply:
BTW, since we're using React 16, we can use the react-popper components as they were intended and drop the need for passing refs around the way we do.

I believe it's something like this.

```
<Manager>
    <Reference>
        ...anchor...
     </Reference>
     <Popper>
        ....portal bubble stuff....
     </Popper>
</Popper.Manager>
```

Full details on the react-popper repo and site.

Reply:
https://github.com/FezVrasta/react-popper#usage